app/controllers/IdentityController.scala

```diff
@@ -41,13 +42,12 @@ class IdentityController(
       .setPasswordGuest(request.body.password, request.body.guestAccountRegistrationToken)
       .fold(
         err => {
-          SafeLogger.error(scrub"Failed to set password: ${err.toString}")
+          SafeLogger.error(scrub"Failed to set password using guest account registration token ${request.body.guestAccountRegistrationToken}: ${err.toString}")
           InternalServerError
         },
         cookies => {
-          SafeLogger.info("Successfully set password")
-          // TODO: add cookies to response header
-          Ok
+          SafeLogger.info(s"Successfully set password using guest account registration token ${request.body.guestAccountRegistrationToken}")
+          Ok.withCookies(CookiesResponse.getCookies(cookies, guardianDomain): _*)
```

Comment:
in terms of naming there's a fair bit of repetition of the word "cookies" going on here but it's not clear what the purpose of the code is: "withCookies(CookiesResponse.getCookies(cookies" but there's not so much about how they relate to each other.
maybe cookies could be identityCookiesResponse and getCookies could be toPlayCookiesList.
so withCookies(CookiesResponse.toPlayCookiesList(identityCookiesResponse
I'm not sure if CookiesResponse makes a great deal of sense but at least this line would make sense in its own context.